src/main/java/org/darisadesigns/polyglotlina/Desktop/CustomControls/DesktopInfoBox.java

```diff
@@ -33,10 +33,12 @@
 import static javax.swing.JOptionPane.DEFAULT_OPTION;
 
 import javax.swing.UIManager;
+import org.darisadesigns.polyglotlina.InfoBox;
 import org.darisadesigns.polyglotlina.PGTUtil;
 
-public class InfoBox extends JFrame {
+public class DesktopInfoBox extends JFrame implements InfoBox {
 
+    private Window parent;
```

Comment:
Moving the parent to here is something that will definitely need to have some thought. I understand 100% the intent here, but that will also prevent PolyGlot from properly centering dialog boxes on particular windows. This doesn't come up a lot in the program since most functionality is limited to a tab style interface, but it's does come up here and there. I'm also concerned this might lead to situations where there's a modal which gets set in an unreachable way. (All of this happens in the POptionPane class, which InfoBox passes the parent window down to).

Reply:
Scratch this. I've seen how you're using the new setup elsewhere, declaring new instances. That makes sense to me.